PWGHF/HFL/TableProducer/electronSelectionWithTPCEMcal.cxx

```diff
@@ -0,0 +1,356 @@
+  HistogramConfigSpec hDeltaPhiDeltaEtaClusterTrackSpecEnergy{HistType::kTHnSparseD, {{400, -0.2, 0.2}, {400, -0.2, 0.2}, {600, -300, 300}, {300, 0.0, 30.0}}};
+  HistogramConfigSpec hPIDSpec{HistType::kTHnSparseD, {{60, 0, 3}, {500, 0.0, 50.0}, {500, 0., 50.}, {300, -15, 15}, {300, 0.0, 30.0}, {400, 0, 2}, {400, 0, 2}}};
+  HistogramConfigSpec hTrackAllInfoSpec{HistType::kTHnSparseD, {{480, 0, 160}, {300, -15, 15}, {500, 0., 50.}, {500, 0., 50.}, {100, -1.5, 1.5}, {100, 0, 7}, {3, 0, 3}}};
+  HistogramConfigSpec hTrackInfoSpec{HistType::kTHnSparseD, {{60, 0, 3}, {480, 0, 160}, {300, -15, 15}, {500, 0., 50.}, {500, 0., 50.}, {100, -1.5, 1.5}, {100, 0, 7}}};
+
+  HistogramRegistry registry{
+    "registry",
+    {{"hNevents", "No of events", {HistType::kTH1F, {{3, 1, 4}}}},
+     {"hZvertex", "z vertex", {HistType::kTH1F, {{100, -100, 100}}}},
+
+     {"hTrackInformation", "Sparse TPC info; dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi;passEMcal;", hTrackAllInfoSpec},
+     {"hClusterInformationBefore", "Cluster Info before match; Energy (GeV);#eta;#varphi", hClusterInfoSpec},
+     {"hClusterInformationAfter", "Cluster Info after match; Energy (GeV);#eta;#varphi", hClusterInfoSpec},
+     {"hPIDafterMatch", "PID Info after match; E/P; dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi;", hTrackInfoSpec},
+     {"hEPRatioafterPID", "E/P Ratio after PID Cuts apply only trackwodca filter", {HistType::kTH2F, {{60, 0, 3}, {100, 0, 10}}}},
+     {"hPIDafterPIDcuts", "PID Info after PID cuts; E/P; #it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});n_{#sigma}^{e};GeV;M02;M20", hPIDSpec},
+     {"hClsTrkEtaPhiDiffTimeEnergy", "ClsTrkEtaPhiDiffTimeEnergy;#Delta#eta;#Delta#varphi;Sec; Energy (GeV)", hDeltaPhiDeltaEtaClusterTrackSpecEnergy}}};
+
+  void init(InitContext&)
+  {
+    registry.get<THnSparse>(HIST("hTrackInformation"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClusterInformationBefore"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClusterInformationAfter"))->Sumw2();
+    registry.get<THnSparse>(HIST("hPIDafterMatch"))->Sumw2();
+    registry.get<THnSparse>(HIST("hPIDafterPIDcuts"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClsTrkEtaPhiDiffTimeEnergy"))->Sumw2();
+  }
+
+  // Track Selection Cut
+  template <typename T>
+  bool selTracks(T const& track)
+  {
+    if (!track.isGlobalTrackWoDCA())
+      return false;
+    if (std::abs(track.dcaXY()) > dcaXYTrackMax || std::abs(track.dcaZ()) > dcaZTrackMax)
+      return false;
+    if (track.eta() < etaTrackMin || track.eta() > etaTrackMax)
+      return false;
+    if ((track.phi() < phiTrackEMCalMin || track.phi() > phiTrackEMCalMax) && (track.phi() < phiTrackDCalMin || track.phi() > phiTrackDCalMax))
+      return false;
+    if (track.pt() < ptTrackMin)
+      return false;
+    return true;
+  }
+
+  // MC Reco Track Selection Cut
+  template <typename T>
+  bool mcSelTracks(T const& mcTrack)
```

Comment:
currently selTracks and mcSelTracks are exactly the same. Such a duplication is not required. I am guessing you would want to include PDG info in the mcSelTracks function. Is that the plan?